target/linux/ath79/dts/qca9563_comfast_cf-e375ac.dts

```diff
@@ -0,0 +1,156 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+#include "qca956x.dtsi"
+
+/ {
+	model = "COMFAST CF-E375AC";
+	compatible = "comfast,cf-e375ac", "qca,qca9563";
+
+	aliases {
+		led-boot = &led_lan;
+		led-failsafe = &led_lan;
+		led-running = &led_lan;
+		led-upgrade = &led_lan;
+	};
+
+	chosen {
+		bootargs = "console=ttyS0,115200n8";
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_lan: led_lan {
+			label = "cf-e375ac:green:lan";
+			gpios = <&gpio 17 GPIO_ACTIVE_HIGH>;
+		};
+
+		led_wlan5g {
+			label = "cf-e375ac:red:wlan5g";
+			gpios = <&gpio 15 GPIO_ACTIVE_HIGH>;
+			linux,default-trigger = "phy0tpt";
+		};
+
+		led_wlan2g {
+			label = "cf-e375ac:blue:wlan2g";
+			gpios = <&gpio 16 GPIO_ACTIVE_HIGH>;
+			linux,default-trigger = "phy1tpt";
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+				linux,code = <KEY_RESTART>;
+				gpios = <&gpio 2 GPIO_ACTIVE_LOW>;
+				debounce-interval = <60>;
+			};
+		};
+
+	watchdog {
+		compatible = "linux,wdt-gpio";
+
+		gpios = <&gpio 6 GPIO_ACTIVE_LOW>;
+		hw_algo = "toggle";
+		hw_margin_ms = <1200>;
+		always-running;
+	};
+};
+
+&uart {
+	status = "okay";
+};
+
+&gpio {
+	status = "okay";
+};
+
+&usb0 {
+	status = "okay";
+};
+
+&pcie {
+	status = "okay";
+};
+
+&spi {
+	status = "okay";
+
+	num-cs = <1>;
+
+	flash@0 {
+		compatible = "winbond,w25q128","jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <25000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x000000 0x040000>;
+				read-only;
+			};
+
+			art: partition@40000 {
+				label = "art";
+				reg = <0x040000 0x010000>;
+				read-only;
+			};
+
+			partition@50000 {
+				compatible = "denx,uimage";
+				label = "firmware";
+				reg = <0x050000 0x0fa0000>;
+			};
+
+			partition@ff0000 {
+				label = "art-backup";
+				reg = <0x0ff0000 0x010000>;
+				read-only;
+			};
+		};
+	};
+};
+
+&mdio0 {
+	status = "okay";
+
+	phy-mask = <0>;
+
+	phy0: ethernet-phy@0 {
+		reg = <0>;
+		phy-mode = "sgmii";
+		qca,mib-poll-interval = <500>;
+
+		qca,ar8327-initvals = <
+			0x04 0x00080080 /* PORT0 PAD MODE CTRL */
+			0x7c 0x0000007e /* PORT0_STATUS */
+		>;
+	};
+};
+
+&eth0 {
+	status = "okay";
+
+	pll-data = <0x03000101 0x00000101 0x00001919>;
+
+	phy-mode = "sgmii";
+	mtd-mac-address = <&art 0x0>;
+	phy-handle = <&phy0>;
+};
+
+&wmac {
+	status = "okay";
+
+	mtd-cal-data = <&art 0x1000>;
+	mtd-mac-address = <&art 0x1002>;
+	mtd-mac-address-increment = <10>;
```

Comment:
Please explain.

Reply:
please check the ART partition dump and original firmware mac assignments in my comment bellow

Reply:
Okay, so since the OEM uses a different assignment that what's in the flash (oh, Comfast ...), I'd actually prefer to calculate all addresses based on art 0x0 (as that's most probably what the vendor firmware actually does), and ignore the other flash locations. Does the device have a MAC address on label.

Please add something like the following to your commit message:

```
MAC addresses:
Though the OEM firmware has four adresses in the usual locations, it appears that the assigned addresses are just incremented in a different way:

interface    address    location
lan:     *:DC     0x0
wan    *:DD    0x1002
2.4g     *:E6    n/a (0x0 + 10)
5g       *:DE     0x6
unused   *:DF   0x5006

Consequently, for the sake of clarity, just use increments of 0x0 for OpenWrt as well.
```